src/foam/core/CompoundException.java

```diff
@@ -32,7 +32,10 @@ public void add(Throwable t) {
 
   public void maybeThrow() {
     if ( exceptions_.size() != 0 ) {
-      throw this;
```

Comment:
`throw this;` makes more sense than looping the exception list then throws the first one.